src/utils/setup/on-create-page.js

```diff
@@ -56,54 +56,57 @@ const getFeaturedArticles = (allArticles, featuredArticleSlugs) => {
     return result;
 };
 
-const getLearnPageFilters = async () => {
-    const filters = {
-        languages: {},
-        products: {},
-    };
+const getLearnPageFilters = allArticles => {
+    const languages = {};
+    const products = {};
 
-    // Get possible language and product values from Stitch
-    const languageValues = await stitchClient.callFunction('getValuesByKey', [
-        metadata,
-        'languages',
-    ]);
-    const productValues = await stitchClient.callFunction('getValuesByKey', [
-        metadata,
-        'products',
-    ]);
-
-    // For each language, build an object with its total count, and count for each product
-    for (let i = 0; i < languageValues.length; i++) {
-        const l = languageValues[i];
-        filters.languages[l._id] = {
-            count: l.count,
-            products: {},
-        };
-        const productValuesForLanguage = await stitchClient.callFunction(
-            'getValuesByKey',
-            [metadata, 'products', { languages: l._id }]
-        );
-        productValuesForLanguage.forEach(pl => {
-            filters.languages[l._id].products[pl._id] = pl.count;
-        });
-    }
-
-    // For each product, build an object with its total count, and count for each language
-    for (let i = 0; i < productValues.length; i++) {
-        const p = productValues[i];
-        filters.products[p._id] = {
-            count: p.count,
-            languages: {},
-        };
-        const languageValuesForProduct = await stitchClient.callFunction(
-            'getValuesByKey',
-            [metadata, 'languages', { products: p._id }]
-        );
-        languageValuesForProduct.forEach(lp => {
-            filters.products[p._id].languages[lp._id] = lp.count;
-        });
-    }
-    return filters;
+    allArticles.forEach(a => {
+        const languagesForArticle = a.query_fields.languages;
+        const productsForArticle = a.query_fields.products;
+        // Go through languages for this article and update filter info
+        if (languagesForArticle) {
+            languagesForArticle.forEach(l => {
+                if (languages[l]) {
+                    languages[l].count++;
+                } else {
+                    languages[l] = {
+                        count: 1,
+                        products: {},
+                    };
+                }
+                if (productsForArticle) {
+                    // If this article also has products, update those counts as well for this language
+                    productsForArticle.forEach(p => {
+                        languages[l].products[p]
+                            ? languages[l].products[p]++
+                            : (languages[l].products[p] = 1);
+                    });
+                }
+            });
+        }
+        // Go through products for this article and update filter info
+        if (productsForArticle) {
```

Comment:
There is some logical redundancy between this conditional and the one on line 67, and so I tried to refactor into a single helper. This got quite complicated and much harder to understand, so I left these separate.

Reply:
Yea I see how these are very similar but the fields within the resulting data structures are where things can get hairy when trying to combine these. Is there a way we can unify the naming for the resulting objects in a way that makes sense and allow us to create a single helper?

Reply:
That's a good point, perhaps something like:
```
languages: {
 count: 1,
 otherTagCounts: {
   stitch: 1,
 }
}
```

The biggest con here is this is clear now, but if we add a third filter it would be a bit harder to understand. I think there might be a way to ratify some of the difference here just not sure what the best way forward is, any thoughts?

Reply:
Yea that makes sense. Hmm I'll have to think about this a little more, especially considering the additional filters the scenario you raised. I'm ok with this moving forward for now though.

Reply:
I'm happy with whatever direction y'all decide, but I'm okay with a bit of redundancy for clarity/extensibility here. As long as it's still straightforward to test.

Reply:
I think at this point having the distinction between languages/products and products/languages supports keeping these a bit separate to understand. We can revisit if we find a great way to refactor, but I think this is the best balance of code clarity and testibility for now